cmd/controller/main.go

```diff
@@ -87,8 +88,13 @@ func main() {
 		HealthProbeBindAddress: fmt.Sprintf(":%d", opts.HealthProbePort),
 	})
 
+	cluster := state.NewCluster(ctx, manager.GetClient())
+	ctx = state.WithClusterState(ctx, cluster)
```

Comment:
SGTM, some of the constructors were getting a bit lengthy with parameters but I don't feel strongly either way.  It can be easier to reason about with it explicitly listed in the constructor.